### DISCUSSION TITLE
单位根 Roots of Unity

### DISCUSSION THREAD

**Discussion by WangQi1111** [zkp-co-learning/zkp-co-learn#116] [Q&A（提问题在隔壁~）] [answered]
1. $F_p^*$为啥一定包含阶数为 $2^\lambda$的乘法子群H
2. 乘法子群H中的元素为啥满足对称性
![image](https://github.com/zkp-co-learning/zkp-co-learn/assets/37581339/f36a7689-e2cb-4b5d-be38-305c4a1a7171)

**Comment by Pupil1999** [comment, marked as answer]
1、根据拉格朗日定理，群元素的阶必定是群阶的因子（如果一个元素阶为12，那么这个元素的4次方的阶就是3）。将群阶素因子分解，可以得到因子2^lambda。
2、((w^(n/2))^2=w^n=1，那么w^(n/2)就是1的二次剩余，如果w^(n/2)等于1，那么这个群的阶就只有n/2，因此不可能为1.只能为-1.

**Comment by Demian101** [comment]
1. 方便 FFT 运算
